framework/framework_test.go

```diff
@@ -506,7 +506,7 @@ func (r *testResource) GetCurrentState(ctx context.Context, obj interface{}) (in
 	if r.CancelingStep == m {
 		canceler, exists := cancelercontext.FromContext(ctx)
 		if exists {
-			canceler <- struct{}{}
+			close(canceler)
```

Comment:
According to a discussion Pawel and I had we decided to close the canceler in case the reconciliation should be canceled for the current custom object. To reflect this the tests of the framework reflect this now.

Reply:
Yeah, cool. Where is the channel init code? I saw it was buffered channel, i.e. `make(chan struct{}, 1)`. It shouldn't be buffered now, as we don't send any message so we don't block, which is good.

Reply:
The initializing code is here: https://github.com/giantswarm/operatorkit/blob/d6088a9cea5f07b8395f7e9d49d8c59032286230/framework/framework.go#L76. We still need to buffer the channel because we try to read from it to know if the canceler was already closed or not. In case the canceler was not closed the checking code would block forever. See https://github.com/giantswarm/operatorkit/blob/d6088a9cea5f07b8395f7e9d49d8c59032286230/framework/cancelercontext/canceler_context.go#L35-L47.

Reply:
I don't think it's true. Buffered channel only prevents from blocking when sending to a channel (as long as there is enough space in the buffer). Receiving from the channel has always the same semantics. This select will always return immediately, doesn't matter the channel is buffered or not.

Reply:
See this https://play.golang.org/p/Z2MK85Rnlh

Also try to comment `default:` line and you'll see deadlock.

Reply:
I see. You are right. Will provide a fix to remove the buffering.